### PR TITLE
Implement NativeStringDelegate

### LINKED ISSUE #2876: iOS. Implement NativeStringDelegate.native.kt
image
(we need to pass locale)

### PR DESCRIPTION
**Pull request by MatkovIvan**
## Testing

Test: added `twoCharactersRepresentedAsSingleUnicodeLetter` (fails on JS because of unimplemented `JsStringDelegate`)

## Issues Fixed

Fixes: https://github.com/JetBrains/compose-multiplatform/issues/2876

### CODE REVIEW

**Review by eymar — APPROVED** [review]

File: compose/ui/ui-text/src/skikoTest/kotlin/androidx/compose/ui/text/StringTest.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+class StringTest {
+
+    @Test
+    fun twoCharactersRepresentedAsSingleUnicodeLetter() {
+        // https://en.wikipedia.org/wiki/D%C5%BE
+        val lowercase = "ǆ" // U+01C6
+        val titlecase = "ǅ" // U+01C5
+        val uppercase = "Ǆ" // U+01C4
+        val serbianLocale = LocaleList("sr")
```
Comment (dima-avdeev-jb) [inline compose/ui/ui-text/src/skikoTest/kotlin/androidx/compose/ui/text/StringTest.kt:41]: can you please add tests for empty String "" too

**Review by dima-avdeev-jb — APPROVED** [review]
